libs/sdk-ui-kit/src/Form/InputWithNumberFormat.tsx

```diff
@@ -0,0 +1,191 @@
+export const MAX_NUMBER = 10 ** 15;
+
+// Max number of digits right to decimal point according to gooddata documentation help.gooddata.com object-datatypes
+export const MAX_DECIMAL_POINT_NUMBERS = 6;
+
+export const DEFAULT_SEPARATORS = {
+    thousand: ",",
+    decimal: ".",
+};
+
+const getDanglingDecimalPointRegExp = memoize((decimal) => new RegExp(`\\${decimal}$`));
+
+const getFormatValidationRegExp = memoize(
+    ({ thousand, decimal }) => new RegExp(`^-?(\\d|\\${thousand})*(\\${decimal}\\d*)?$`),
+);
+
+const parseStandardNumberString = (numberString: string) => {
+    const belowDecimal = numberString.split(".")[1];
+
+    const roundedNumberString =
+        belowDecimal && belowDecimal.length >= MAX_DECIMAL_POINT_NUMBERS
+            ? parseFloat(numberString).toFixed(MAX_DECIMAL_POINT_NUMBERS)
+            : numberString;
+
+    const number = parseFloat(roundedNumberString);
+
+    return number === 0 ? 0 : number;
+};
+
+const convertFormattedStringToStandard = (formattedString: string, { thousand, decimal }: Separators) => {
+    const withoutThousandSeparators = formattedString.toString().split(thousand).join("");
+
+    const withoutDanglingDecimalPoint = withoutThousandSeparators.replace(
+        getDanglingDecimalPointRegExp(decimal),
+        "",
+    );
+
+    const withStandardDecimalPoint = withoutDanglingDecimalPoint.split(decimal).join(".");
+
+    return withStandardDecimalPoint.length > 0 ? withStandardDecimalPoint : null;
+};
+
+const parse = (value: any | string, separators: Separators) => {
+    if (value === null || value === "" || value === "-") {
+        return null;
+    }
+
+    const numberString = convertFormattedStringToStandard(value, separators);
+
+    if (numberString === null) {
+        return null;
+    }
+
+    return parseStandardNumberString(numberString);
+};
+
+const isValid = (value: any, separators: Separators) => {
+    return (
+        getFormatValidationRegExp(separators).test(value) && Math.abs(parse(value, separators)) <= MAX_NUMBER
+    );
+};
+
+const format = (value: any, { thousand, decimal }: Separators) => {
+    if (value === null) {
+        return "";
+    }
+
+    const [aboveDecimal, belowDecimal] = value.toString().split(".");
+
+    const aboveDecimalFormatted = aboveDecimal.replace(/(\d)(?=(\d{3})+(?!\d))/g, `$1${thousand}`);
+
+    return belowDecimal ? `${aboveDecimalFormatted}${decimal}${belowDecimal}` : aboveDecimalFormatted;
+};
+/**
+ * @internal
+ */
+export interface InputWithNumberFormatOwnProps {
+    separators: Separators;
+}
+
+/**
+ * @internal
+ */
+
+export interface InputWithNumberFormatState {
+    value: number;
+    isFocused: boolean;
+}
+
+/**
+ * @internal
+ */
+
+export type InputWithNumberFormatProps = InputWithNumberFormatOwnProps & InputPureProps;
+
+/**
+ * @internal
+ */
+
+export class InputWithNumberFormat extends React.PureComponent<
+    InputWithNumberFormatProps,
+    InputWithNumberFormatState
+> {
+    private input: InputPure;
```

Comment:
Can we please use `createRef`? The callback refs were deprecated and this would be a good opportunity to make this change (it was too risky for goodstrap).

Reply:
these refs are leaking and used outside the component